### PR TITLE
[API] Change release API paths

### PR DESCRIPTION
**Pull request by andrebruch**
Corrects the following release API routes:

- `GET /repos/{owner}/{repo}/releases/{id}/assets/{attachment_id}` 
- -> `GET /repos/{owner}/{repo}/releases/assets/{attachment_id}`
- `PATCH /repos/{owner}/{repo}/releases/{id}/assets/{attachment_id}` 
- -> `PATCH /repos/{owner}/{repo}/releases/assets/{attachment_id}`
- `DELETE /repos/{owner}/{repo}/releases/{id}/assets/{attachment_id}` 
- -> `DELETE /repos/{owner}/{repo}/releases/assets/{attachment_id}`

The routes were in conflict with the Github API:

- [GET /repos/{owner}/{repo}/releases/assets/{asset_id}](https://docs.github.com/en/rest/reference/repos#get-a-release-asset)
- [PATCH /repos/{owner}/{repo}/releases/assets/{asset_id}](https://docs.github.com/en/rest/reference/repos#update-a-release-asset)
- [DELETE /repos/{owner}/{repo}/releases/assets/{asset_id}](https://docs.github.com/en/rest/reference/repos#delete-a-release-asset)

The previous routes are marked as deprecated, but have not been removed.

### CODE REVIEW

**Comment by 6543** [comment]
I would not say it's a fix but an enhancement. since endpoints work just fine. only github had made some changes we are adopting with this ...

File: routers/api/v1/repo/release_attachment.go [hunk 1]
```diff
@@ -35,6 +35,50 @@ func GetReleaseAttachment(ctx *context.APIContext) {
...
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/Attachment"
+
+	attachID := ctx.ParamsInt64(":asset")
+	attach, err := models.GetAttachmentByID(attachID)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "GetAttachmentByID", err)
+		return
+	}
```
Comment (6543) [inline routers/api/v1/repo/release_attachment.go:55]: with this i can delete attachments i should not have the rights to do so ... you have to test if attachment you get belongs to this repo ...

Comment (6543) [reply routers/api/v1/repo/release_attachment.go:55]: & tests for this are needed + some general tests too

**Comment by 6543** [comment]
`make generate-swagger` is missing & some tests :)

**Comment by andrebruch** [comment]
`make generate-swagger` was not missing ;)
On my system, the command did not append an extra line to the `v1_json.tmpl` file. 
After I manually inserted an empty line, the build is passing.

That was the error:
```
Please run 'make generate-swagger' and commit the result:
diff --git a/templates/swagger/v1_json.tmpl b/templates/swagger/v1_json.tmpl
index e217c174f..5da925a10 100644
--- a/templates/swagger/v1_json.tmpl
+++ b/templates/swagger/v1_json.tmpl
@@ -17070,4 +17070,4 @@
       "TOTPHeader": []
     }
   ]
-}
\ No newline at end of file
make: *** [Makefile:255: swagger-check] Error 1
+}

```

File: routers/api/v1/repo/release_attachment.go [hunk 2]
```diff
@@ -276,9 +412,74 @@ func EditReleaseAttachment(ctx *context.APIContext) {
...
+
+	ownerName := ctx.Params(":username")
+	repoName := ctx.Params(":reponame")
+	attachID := ctx.ParamsInt64(":asset")
+	attach, err := models.GetAttachmentByID(attachID)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "GetAttachmentByID", err)
+		return
+	}
+	repo, err := models.GetRepositoryByOwnerAndName(ownerName, repoName)
```
Comment (6543) [inline routers/api/v1/repo/release_attachment.go:449]: you can get repo from context!

**Review by 6543 — CHANGES_REQUESTED** [review]

**Comment by 6543** [comment]
also we could dedublicate some code

**Comment by wxiaoguang** [comment]
I guess Gitea's API has been different from GitHub's for long time.

Is this PR still related?

**Pull request closed without merging**